### PR TITLE
Implement Future for Context instead of TryFuture

### PR DESCRIPTION
**Pull request by pluth**
Implements the changes discussed in #134 
Implements Future for try_future::Context and try_future::WithContext instead of TryFuture. TryFuture implementation should still be available as it was through a blanket implementation in futures_core. Same thing for the try_stream module.

I've extended the futures compatibility test to test for the Future implementation (very uncreatively) and removed two into_stream() calls to demonstrate Stream implementation.

### CODE REVIEW

File: compatibility-tests/futures/src/lib.rs [hunk 1]
```diff
@@ -61,14 +61,29 @@ async fn load_stock_data_concurrent() -> Result<String, Error> {
     Ok(format!("{}+{}", apple, google))
 }
 
+// Returns of the combinators implement `Future`
```
Comment (shepmaster) [inline compatibility-tests/futures/src/lib.rs]:
```suggestion
// Return values of the combinators implement `Future`
```

File: compatibility-tests/futures/src/lib.rs [hunk 2]
```diff
@@ -61,14 +61,29 @@ async fn load_stock_data_concurrent() -> Result<String, Error> {
...
+}
+
 // Can be used as a `Stream` combinator
 async fn load_stock_data_series() -> Result<String, Error> {
     let apple = api::keep_fetching_page("apple").context(UnableToLoadAppleStock);
     let google = api::keep_fetching_page("google").with_context(|| UnableToLoadGoogleStock {
         name: String::from("stream"),
     });
 
-    let together = apple.into_stream().zip(google.into_stream());
```
Comment (shepmaster) [inline compatibility-tests/futures/src/lib.rs]: I'm not following why this change was made; was it *required* to change or is it simply clearer to change it?

File: src/futures/try_stream.rs [hunk 3]
```diff
@@ -3,6 +3,7 @@
 //! [`TryStream`]: futures_core::TryStream
 
 use futures_core::stream::TryStream;
+use futures_core::stream::Stream;
```
Comment (shepmaster) [inline src/futures/try_stream.rs]: Let's go ahead and do `use futures_core::stream::{Stream, TryStream};`

File: compatibility-tests/futures/src/lib.rs (hunk 2 above)
Comment (shepmaster) [reply compatibility-tests/futures/src/lib.rs]: Ah, right. `into_stream` is the hack needed to convert a `TryStream` back into a `Stream`, which we don't need anymore.

However, leaving the method calls *would* still work, since we implicitly implement `TryStream`.

**Comment by shepmaster** [comment]
Thanks!

**Comment by pluth** [comment]
Thanks for fixing and implementing it. For me, it can wait. As we discussed, I can just work with the Result combinators

Thanks again

**Comment by shepmaster** [comment]
> For me, it can wait

Too bad! I just released 0.4.4 with this and another fix 😉
